### PR TITLE
remove MachineRef when vm is deleted

### LINKED ISSUE #551: machines and vspheremachine show up even after deleting
/kind bug

**What steps did you take and what happened:**
I created a management cluster and a workload cluster on vSphere.
Applied a `machinedeployment` with 1 replica on the management cluster. The VM gets created on vSphere. Then, the `machinedeployment` was deleted. 
Next, a `machinedeployment` with 2 replicas is applied on the workload cluster. After both the vm are created the `machinedeployment` is deleted. The VM get destroyed form vSphere.

At the end a `machinedeployment` with 2 replicas is applied again on the workload cluster.

Now if we run `kubectl get machines`, the machines still show up in "deleting" state and remain in the state indefinitely.
```
➜ kubectl --kubeconfig kubeconfig get machines
NAME                                       PROVIDERID                                       PHASE
management-cluster-controlplane-0          vsphere://42119170-11a2-c0f1-5522-3f10c4e8e62a   running
management-cluster-md-0-84455f76d4-r9t28   vsphere://42112a9d-3e09-1717-8c15-e60150e28481   deleting
workload-cluster-controlplane-0            vsphere://4211692f-2e55-aa58-e69c-7a7ed1914812   running
workload-cluster-md-0-59489b94fd-6zqpf     vsphere://4211cb55-7db5-34ba-a079-b1f2b81b1b16   deleting
workload-cluster-md-0-59489b94fd-8wmrj     vsphere://421196f4-d807-3c9f-db1e-1480216ac66e   running
workload-cluster-md-0-59489b94fd-h72g7     vsphere://421185b1-45c3-096b-912c-060a536b9210   running
workload-cluster-md-0-59489b94fd-m6qlw                                                      deleting
```

Similarly, if we do `kubectl get vspheremachines` the vspheremachines that were deleted still show up.
```
➜ kubectl --kubeconfig kubeconfig get vspheremachines
NAME                                AGE
management-cluster-controlplane-0   44m
management-cluster-md-0-n2cvb       41m
workload-cluster-controlplane-0     34m
workload-cluster-md-0-fdl7h         25m
workload-cluster-md-0-mcmfz         31m
workload-cluster-md-0-mmmw6         31m
workload-cluster-md-0-xwxjs         25m
```

The vspheremachines 
-`workload-cluster-md-0-mcmfz`
-`workload-cluster-md-0-mmmw6` 
belong to the deleted `machinedeployment` on the workload-cluster. 

The vspheremachine `management-cluster-md-0-n2cvb` belongs to the `machinedeployment` on he management-cluster but all of them still show up.

**What did you expect to happen:**
The deleted machines and vspheremachines should be completely removed.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**
The manifests were generated using: `gcr.io/cluster-api-provider-vsphere/release/manifests@sha256:14daace258e6da4f13482eb8e3e125bb0dee787bd4666c382781ad38337cbaf1`

### PR DESCRIPTION
**Pull request by ykakarap**
**What this PR does / why we need it**:
This PR adds the logic to remove the `MachineRef` if a vm does not exist.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #551 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by ykakarap** [comment]
@akutz I did not refactor the `hasInFlightTask` instead I added the needed check in `DeleteVM`. Refactoring the `hasInFlightTask` will need modifications in the `ReconcileVM` and I thought it was best to leave it as is for now.

**Comment by akutz** [comment]
Hi @andrewsykim,

Are we sure `WithValues` is the same as `WithName`? Per the [GoDocs](https://godoc.org/github.com/go-logr/logr):

```golang
    // WithValues adds some key-value pairs of context to a logger.
    // See Info for documentation on how key/value pairs work.
    WithValues(keysAndValues ...interface{}) Logger

    // WithName adds a new element to the logger's name.
    // Successive calls with WithName continue to append
    // suffixes to the logger's name.  It's strongly reccomended
    // that name segments contain only letters, digits, and hyphens
    // (see the package documentation for more information).
    WithName(name string) Logger
```

It sure seems like `WithName` has a different distinction. But, then again, the go-logr `Logger` interface is just that, an interface. Looking at an implementation in the controller-runtime's [`DelegatingLogger`](https://github.com/kubernetes-sigs/controller-runtime/blob/ab6131a999ca9e6ff5bf83dba70dc7751751135f/pkg/log/deleg.go#L36-L62), it also appears that `WithName` initializes separate fields than the values:

```golang
// WithName provides a new Logger with the name appended
func (p *loggerPromise) WithName(l *DelegatingLogger, name string) *loggerPromise {
	res := &loggerPromise{
		logger:       l,
		name:         &name,
		promisesLock: sync.Mutex{},
	}

	p.promisesLock.Lock()
	defer p.promisesLock.Unlock()
	p.childPromises = append(p.childPromises, res)
	return res
}

// WithValues provides a new Logger with the tags appended
func (p *loggerPromise) WithValues(l *DelegatingLogger, tags ...interface{}) *loggerPromise {
	res := &loggerPromise{
		logger:       l,
		tags:         tags,
		promisesLock: sync.Mutex{},
	}

	p.promisesLock.Lock()
	defer p.promisesLock.Unlock()
	p.childPromises = append(p.childPromises, res)
	return res
}
```

File: pkg/cloud/vsphere/services/govmomi/service.go [hunk 1]
```diff
@@ -124,6 +124,22 @@ func (vms *VMService) DestroyVM(ctx *context.MachineContext) (infrav1.VirtualMac
...
 	}
 
+	// check if VM actually exists
+	if ctx.VSphereMachine.Spec.MachineRef != "" {
+		moRefID, err := findVMByInstanceUUID(ctx)
+		if err != nil {
+			return vm, err
+		}
+		if moRefID == "" {
+			// No vm exits
```
Comment (akutz) [inline pkg/cloud/vsphere/services/govmomi/service.go]: Hi @ykakarap,

Can you please correct the spelling to "exists"?

**Pull request merged**